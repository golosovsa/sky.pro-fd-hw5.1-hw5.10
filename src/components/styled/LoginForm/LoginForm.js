import styled from "styled-components";
import {
    layoutSizeMixin,
    calcWidth,
    calcHeight,
} from "../../../globals/layout/index";
import { applyThemeMixin } from "../../../globals/theme/index";

const LoginForm = styled.form`
    ${applyThemeMixin("login", "form")}
    ${layoutSizeMixin(366, 439)}
    padding: ${calcHeight(45)} ${calcWidth(44)};
    border-radius: min(${calcWidth(12)}, ${calcHeight(12)});
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-between;
    align-items: center;
    align-content: center;
`;

export default LoginForm;
