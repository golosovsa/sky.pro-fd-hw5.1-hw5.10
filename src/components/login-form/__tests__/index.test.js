import ShallowRenderer from 'react-test-renderer/shallow'
import LoginForm from '..'

describe('<LoginForm />', () => {
  describe('Snapshot test', () => {
    it('should render successfully and match snapshot', () => {
      const renderer = new ShallowRenderer()
      const tree = renderer.render(<LoginForm />)

      expect(tree).toMatchSnapshot()
    })

    it('should render inputs and buttons and disabled successfully and match snapshot', () => {
      const renderer = new ShallowRenderer()
      const tree = renderer.render(<LoginForm inputs={123} buttons={123} disabled={true} />)

      expect(tree).toMatchSnapshot()
    })
  })
})
