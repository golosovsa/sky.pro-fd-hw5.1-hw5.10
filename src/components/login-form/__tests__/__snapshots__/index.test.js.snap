// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> Snapshot test should render inputs and buttons and disabled successfully and match snapshot 1`] = `
<styled.div
  disabled={true}
>
  <styled.form
    onSubmit={[Function]}
  >
    <styled.div>
      <Styled(Component) />
    </styled.div>
    <styled.fieldset
      disabled={true}
    >
      123
    </styled.fieldset>
    <styled.fieldset
      disabled={true}
    >
      123
    </styled.fieldset>
  </styled.form>
</styled.div>
`;

exports[`<LoginForm /> Snapshot test should render successfully and match snapshot 1`] = `
<styled.div
  disabled={false}
>
  <styled.form
    onSubmit={[Function]}
  >
    <styled.div>
      <Styled(Component) />
    </styled.div>
    <styled.fieldset
      disabled={false}
    />
    <styled.fieldset
      disabled={false}
    />
  </styled.form>
</styled.div>
`;
